bootstrap/api/swagger.yaml

```diff
@@ -0,0 +1,317 @@
+schemes:
+- "https"
+paths:
+  /projects/{project}/configs:
+    post:
+      tags:
+        - "config"
+      summary: "Create a new kubeflow config entry on source repo"
+      security:
+        - BearerToken: []
+      operationId: "createConfig"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "body"
+          name: "body"
+          description: "Config needs to be created"
+          required: true
+          schema:
+            $ref: "#/definitions/KfConfig"
+      responses:
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+    get:
+      tags:
+        - "config"
+      summary: "List kubeflow config entries in target project"
+      security:
+        - BearerToken: []
+      operationId: "listConfigs"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+      responses:
+        200:
+          description: "Kubeflow config entries"
+          schema:
+            $ref: "#/definitions/KfConfigEntries"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+  /projects/{project}/configs/{config}/version/{version}/file:
+    get:
+      tags:
+        - "config"
+      summary: "Get kubeflow config entry file"
+      security:
+        - BearerToken: []
+      operationId: "getConfigFile"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "config"
+          description: "config entry name"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "version"
+          description: "config entry kubeflow version"
+          required: true
+          type: "string"
+        - in: "query"
+          name: "filepath"
+          description: "file relative path"
+          required: true
+          type: "string"
+      responses:
+        200:
+          description: "Kubeflow config entries"
+          schema:
+            $ref: "#/definitions/KfConfigFile"
+        404:
+          description: "Entry not found"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+    put:
+      tags:
+        - "config"
+      description: "Update kubeflow config entry file"
```

Comment:
Why do we need a put operation on config?

If people want to update the config couldn't they do this directly by modifying the file; e.g. in the source repo?